pkg/loadbalancer/loadbalancer.go

```diff
@@ -475,17 +475,21 @@ func (a *L3n4Addr) DeepCopy() *L3n4Addr {
 	}
 }
 
-// Hash calculates L3n4Addr's internal SHA256Sum.
+// Hash calculates a unique string of the L3n4Addr e.g for use as a key in maps
 func (a L3n4Addr) Hash() string {
+	// Note: 32 bytes is not enough for long IPv6 addresses, but it is cheaper
+	// to reallocate on overflow than to check if a.IP is IPv6
+	b := make([]byte, 0, 32)
+
+	b = append(b, a.IP.String()...)
+	b = append(b, '|')
 	// FIXME: Remove Protocol's omission once we care about protocols.
-	protoBak := a.Protocol
-	a.Protocol = ""
-	defer func() {
-		a.Protocol = protoBak
-	}()
-
-	str := []byte(fmt.Sprintf("%+v", a))
-	return fmt.Sprintf("%x", sha512.Sum512_256(str))
+	b = append(b, '|')
+	b = strconv.AppendUint(b, uint64(a.Port), 10)
+	b = append(b, '|')
+	b = strconv.AppendUint(b, uint64(a.Scope), 10)
+
+	return string(b)
```

Comment:
One more thing, how does this relate to L3n4AddrID? E.g. https://github.com/cilium/cilium/blob/2b9485df05ec24fe051684ce992a82dabd0b9e56/pkg/maps/lbmap/lbmap.go#L563 is calling the Hash() from an L3n4AddrID type where the Sprintf() was on %+v.

Reply:
Calling `fe.Hash()` (where `fe` is of type `L3n4AddrID`) is equivalent to calling `fe.L3n4Addr.Hash()`. In other words, there is no inheritance. The `%+v` part was still only printing the contents of `L3n4Addr`, the `ID` is and was omitted. 

Not sure that is intended, but this commit does not change that.

Reply:
`L3n4AddrID.Hash` is simply `L3n4Addr.Hash` and thus the `ID` field will not be used. If that is an issue, we probably want to define `L3n4AddrID.Hash` as `L3n4Addr.Hash()|ID` or something along those lines.